### PR TITLE
[native_assets_cli] Use JSON instead of YAML

### PR DESCRIPTION
**Pull request by dcharkes**
* https://github.com/dart-lang/native/issues/991

### CODE REVIEW

**Comment by mosuem** [comment]
Should this rather be based of the move-to-`hook/` PR?

**Comment by dcharkes** [comment]
> Should this rather be based of the move-to-`hook/` PR?

Yeah, I tried to do that. but then the CI didn't trigger, so I based it on main again... 🤷

File: pkgs/native_assets_cli/lib/src/model/asset.dart [hunk 1]
```diff
@@ -5,19 +5,19 @@
...
 abstract final class AssetImpl implements Asset {
-  Map<String, Object> toYaml(Version version);
+  Map<String, Object> toJson(Version version);
 
-  static List<AssetImpl> listFromYamlList(YamlList list) {
+  static List<AssetImpl> listFromJsonList(List<Object?> list) {
     final assets = <AssetImpl>[];
     for (final yamlElement in list) {
-      final yamlMap = as<YamlMap>(yamlElement);
+      final yamlMap = as<Map<Object?, Object?>>(yamlElement);
```
Comment (mosuem) [inline pkgs/native_assets_cli/lib/src/model/asset.dart:13]: Variable names are still `yaml`

File: pkgs/native_assets_cli/lib/src/model/build_output.dart [hunk 2]
```diff
@@ -48,12 +48,19 @@ final class BuildOutputImpl implements BuildOutput {
...
+    } else {
+      // TODO(https://github.com/dart-lang/native/issues/1000): At some point
+      // remove the YAML fallback.
+      json = loadYaml(jsonString);
+    }
+    return BuildOutputImpl.fromJson(as<Map<Object?, Object?>>(json));
   }
 
-  factory BuildOutputImpl.fromYaml(YamlMap yamlMap) {
+  factory BuildOutputImpl.fromJson(Map<Object?, Object?> yamlMap) {
```
Comment (mosuem) [inline pkgs/native_assets_cli/lib/src/model/build_output.dart:63]: Again variable names - a global search for `yaml` should do the trick to find all instances :)

File: pkgs/native_assets_cli/test/model/asset_test.dart [hunk 3]
```diff
@@ -156,16 +157,16 @@ void main() {
 
   test('asset yaml', () {
     final yaml = yamlEncode([
-      for (final item in assets) item.toYaml(BuildOutputImpl.latestVersion)
+      for (final item in assets) item.toJson(BuildOutputImpl.latestVersion)
```
Comment (mosuem) [inline pkgs/native_assets_cli/test/model/asset_test.dart:160]: So the asset list still is a YAML?

**Review by mosuem — CHANGES_REQUESTED** [review]

File: pkgs/native_assets_cli/test/model/asset_test.dart (hunk 3 above)
Comment (dcharkes) [reply pkgs/native_assets_cli/test/model/asset_test.dart:160]: No, this test is testing what if we encode the JSON object (not string) to YAML. I find that easier to test.

Comment (mosuem) [reply pkgs/native_assets_cli/test/model/asset_test.dart:160]: That's personal preference then :) I find JSON to be very readable (if nicely formatted).

File: pkgs/native_assets_builder/lib/src/model/kernel_assets.dart [hunk 4]
```diff
@@ -6,7 +6,7 @@
 /// kernel file.
 ///
 /// The `native_assets.yaml` embedded in a kernel file has a different format
```
Comment (mosuem) [inline pkgs/native_assets_builder/lib/src/model/kernel_assets.dart:8]: `native_assets.yaml`?

**Review by mosuem — APPROVED** [review]

File: pkgs/native_assets_builder/lib/src/model/kernel_assets.dart (hunk 4 above)
Comment (dcharkes) [reply pkgs/native_assets_builder/lib/src/model/kernel_assets.dart:8]: yep, that's not part of the protocol it's internal to the native_assets_builder and the CFE.

Maybe we should consider moving that to JSON as well. If so, in a separate PR.

**Pull request merged**